pkg/operator/controllers/previewfeature/nsgflowlogs/nsgflowlogs.go

```diff
@@ -0,0 +1,89 @@
+package nsgflowlogs
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	aropreviewv1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/preview.aro.openshift.io/v1alpha1"
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-08-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/sirupsen/logrus"
+	"time"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/network"
+)
+
+const (
+	networkWatcherResourceGroupName = "NetworkWatcherRG"
+)
+
+func NewFeature(flowLogsClient network.FlowLogsClient, log *logrus.Entry) *nsgFlowLogsFeature {
+	return &nsgFlowLogsFeature{
+		log:            log,
+		flowLogsClient: flowLogsClient,
+	}
+}
+
+type nsgFlowLogsFeature struct {
+	log *logrus.Entry
+
+	flowLogsClient network.FlowLogsClient
+}
+
+func (n *nsgFlowLogsFeature) Name() string {
+	return "nsgFlowLogsFeature"
+}
+
+func (n *nsgFlowLogsFeature) Reconcile(ctx context.Context, instance *aropreviewv1alpha1.PreviewFeature) error {
+	if instance.Spec.NSGFlowLogs == nil {
+		return nil
+	}
+
+	if !instance.Spec.NSGFlowLogs.Enabled {
+		return n.Disable(ctx, instance)
+	}
+
+	return n.Enable(ctx, instance)
+}
+
+func (n *nsgFlowLogsFeature) Enable(ctx context.Context, instance *aropreviewv1alpha1.PreviewFeature) error {
+	// build a request as described here https://docs.microsoft.com/en-us/azure/network-watcher/network-watcher-nsg-flow-logging-rest#enable-network-security-group-flow-logs
+	flowLog := mgmtnetwork.FlowLog{
+		Location: to.StringPtr(instance.Spec.NSGFlowLogs.Location),
```

Comment:
Should it ibe the same as cluster location?

Reply:
It is, and we can get rid of it if we take the network watcher id instead of relying on the conventional one existing